src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React from 'react';
+import { Todo } from '../../types/Todo';
+import classNames from 'classnames';
+import * as todosService from '../../api/todos';
+
+type Props = {
+  todo: Todo;
+  todos: Todo[];
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setErrorMessage: (errorMessage: string) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo: { title, completed, id },
+  todos,
+  setTodos,
+  setErrorMessage,
+}) => {
+  const handleToggleComplete = (todoId: number) => {
+    const updatedTodos = todos.map(todo => {
+      if (todo.id === todoId) {
+        return { ...todo, completed: !todo.completed };
+      }
+
+      return todo;
+    });
+
+    setTodos(updatedTodos);
+  };
+
+  function removeTodo(todoId: number) {
+    setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+
+    return todosService
+      .deleteTodo(todoId)
+      .then(() => {})
+      .catch(error => {
+        setErrorMessage(`Unable to delete a todo`);
+        throw error;
```

Comment:
do not throw an error here - you handled it already, so no need to throw it again. It will lead to a page crash (because you don't try to catch the error thrown here somewhere else)